manifests/init.pp

```diff
@@ -26,141 +26,143 @@
-  String $proxy_cache_inactive                      = '20m',
-  String $proxy_cache_keys_zone                     = 'd2:100m',
-  String $proxy_cache_levels                        = '1',
-  String $proxy_cache_max_size                      = '500m',
-  Optional[Variant[Hash, String]] $proxy_cache_path = undef,
-  Optional[Integer] $proxy_cache_loader_files       = undef,
-  Optional[String] $proxy_cache_loader_sleep        = undef,
-  Optional[String] $proxy_cache_loader_threshold    = undef,
-  Optional[Enum['on', 'off']] $proxy_use_temp_path  = undef,
-  $proxy_connect_timeout                            = '90',
-  Integer $proxy_headers_hash_bucket_size           = 64,
-  Optional[String] $proxy_http_version              = undef,
-  $proxy_read_timeout                               = '90',
-  $proxy_redirect                                   = undef,
-  $proxy_send_timeout                               = '90',
-  Array $proxy_set_header                           = [
+  $accept_mutex                                              = 'on',
+  $accept_mutex_delay                                        = '500ms',
+  $client_body_buffer_size                                   = '128k',
+  String $client_max_body_size                               = '10m',
+  $client_body_timeout                                       = '60',
+  $send_timeout                                              = '60',
+  $lingering_timeout                                         = '5',
+  Optional[String] $events_use                               = undef,
+  String $fastcgi_cache_inactive                             = '20m',
+  Optional[String] $fastcgi_cache_key                        = undef,
+  String $fastcgi_cache_keys_zone                            = 'd3:100m',
+  String $fastcgi_cache_levels                               = '1',
+  String $fastcgi_cache_max_size                             = '500m',
+  Optional[String] $fastcgi_cache_path                       = undef,
+  Optional[String] $fastcgi_cache_use_stale                  = undef,
+  $gzip                                                      = 'on',
+  $gzip_buffers                                              = undef,
+  $gzip_comp_level                                           = 1,
+  $gzip_disable                                              = 'msie6',
+  $gzip_min_length                                           = 20,
+  $gzip_http_version                                         = 1.1,
+  $gzip_proxied                                              = 'off',
+  $gzip_types                                                = undef,
+  $gzip_vary                                                 = 'off',
+  Optional[Variant[Hash, Array]] $http_cfg_prepend           = undef,
+  Optional[Variant[Hash, Array]] $http_cfg_append            = undef,
+  Optional[Variant[Array[String], String]] $http_raw_prepend = undef,
+  Optional[Variant[Array[String], String]] $http_raw_append  = undef,
```

Comment:
Hi @wyardley  -- thanks for the comments! If a String is passed in, the template will cast it into an Array with one member e.g.

```
[1] pry(main)> v = 'String'
=> "String"
[2] pry(main)> Array(v)
=> ["String"]
```
I just used the added flexibility that was already at https://github.com/voxpupuli/puppet-nginx/blob/360e700d1aa98bdf9bb05f1379becdc01ed001ea/templates/server/location_header.erb#L67